src/lib/Theme.ts

```diff
@@ -943,7 +947,42 @@ export default class Theme {
             theme.config = theme.config || {};
             theme.config.global_schema = globalConfigSchema;
             theme.config.current = globalConfigData.current || 'default';
-            theme.config.list = globalConfigData.list || [{ name: 'default' }];
+
+            // Modify list to update deleted page's prop
+            let newList = null;
+            if (globalConfigData.list && Object.keys(allowedDefaultProps).length > 0) {
+                newList = globalConfigData.list.map(listItem => {
+                    if (!listItem.page) return listItem
+
+                    // delete extra props from all list (Default, Blue, Dark)
+                    const page = listItem.page.map(pageData => {
+                        // allowedDefaultProps object have deleted page name as key
+                        // If current page is not deleted page, then no changes needed
+                        if (!allowedDefaultProps[pageData.page]) return pageData;
+
+                        const { settings } = pageData;
+                        const newSettingsProps = settings.props
+
+                        Object.keys(newSettingsProps).forEach(prop => {
+                            if (!allowedDefaultProps[pageData.page].includes(prop)) {
+                                delete newSettingsProps[prop]
+                            }
+                        })
+                        return {
+                            ...pageData,
+                            settings: {
+                                ...pageData.settings,
+                                props: newSettingsProps
+                            }
+                        }
+                    })
+                    return {
+                        ...listItem,
+                        page
+                    }
```

Comment:
```suggestion
                    return page;
```

Reply:
We are iterating list item, and page is inside list item. So while returning values, we need to return whole list item no?